### PR TITLE
xgboost 2.1.1

### PR DESCRIPTION
**Pull request by lorepirri**
xgboost 2.1.1

**Destination channel:** defaults

### Links

- [PKG-5352]
- dev_url (pypi): https://github.com/dmlc/xgboost/tree/v2.1.1
- [v2.0.3...v2.1.1](https://github.com/dmlc/xgboost/compare/v2.0.3...v2.1.1) diff: https://github.com/dmlc/xgboost/compare/v2.0.3...v2.1.1
- conda-forge: https://github.com/conda-forge/xgboost-feedstock/blob/main/recipe/meta.yaml
- diff 2.03…2.1.1: https://github.com/conda-forge/xgboost-feedstock/compare/5c1e0fe825acc654cb9c65adad76814446fd11bc...9753ed30fc17610bf34ba475b9d117b6b85bc35f
- pypi: https://pypi.org/project/xgboost/2.1.1
- pypi inspector: https://inspector.pypi.io/project/xgboost/2.1.1

### Explanation of changes:

- bump version to `2.1.1`
- set right folders:
  - cpp src is now in the root instead of `src_cpp`.
  - python is in the `python-package` folder instead of the root.
- [update to vs2022](https://github.com/AnacondaRecipes/xgboost-feedstock/pull/19/commits/dbd4a87d0432ac45e54880d622265c77b4331890) because of `error LNK2001: unresolved external symbol _guard_check_icall_$fo$`
- ~add `nccl` dependency for linux, I am not sure if it's needed for `s390x` it's not there, `s390x` skipped but I am not sure it is a good idea.~ remove `nvidia-nccl-cu12` dependency as it is only for CUDA builds. See comment https://github.com/AnacondaRecipes/xgboost-feedstock/pull/19#pullrequestreview-2230352365
- there are tests in the archive, but I will stick with the one we had before.

[PKG-5352]: https://anaconda.atlassian.net/browse/PKG-5352?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by cbouss — COMMENTED** [review]
nccl is for if you are building the cuda variant.
So far we've only built the cpu variant. 

As long as you don't set these options to ON it will not build for cuda:
https://github.com/dmlc/xgboost/blob/v2.1.1/CMakeLists.txt#L75-L80

Just apply your patch to all architectures and you should be good.

**Comment by lorepirri** [comment]
> nccl is for if you are building the cuda variant. So far we've only built the cpu variant.
> 
> As long as you don't set these options to ON it will not build for cuda: https://github.com/dmlc/xgboost/blob/v2.1.1/CMakeLists.txt#L75-L80
> 
> Just apply your patch to all architectures and you should be good.

Thank you @cbouss !

File: recipe/meta.yaml [hunk 1]
```diff
@@ -68,8 +73,11 @@ outputs:
         - python
         - numpy
         - scipy
+        - nccl         # [linux and not aarch64]
```
Comment (danpetry) [inline recipe/meta.yaml]:
```suggestion
```

**Review by danpetry — APPROVED** [review]
abs.yaml can be deleted

**Review by cbouss — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (lorepirri) [reply recipe/meta.yaml]: done

Comment (lorepirri) [reply recipe/meta.yaml]: done

**Comment by lorepirri** [comment]
> abs.yaml can be deleted

I tested the idea but it did not work. See logs: https://github.com/AnacondaRecipes/xgboost-feedstock/pull/19/commits/9d6dc6e0fa56e330730f1753233bd4b6d4b52cb0

**Comment by danpetry** [comment]
Ah right, that must be to do with skipping existing and multi-output feedstocks not working properly together

**Comment by lorepirri** [comment]
After merging `osx-arm64` does not pass the test phase on `py312` with `xgboost 2.1.1 is not supported on this platform`. Something might have changed on the dependencies because also locally now it does not build.

**Comment by lorepirri** [comment]
It's the `pip check` on  `pip 24.2` https://github.com/pypa/pip/issues/12884

**Pull request merged**